src/backend/cuda/subconfigure.m4

```diff
@@ -58,17 +58,28 @@ AC_ARG_WITH([cuda-sm],
 # --with-cuda
 PAC_SET_HEADER_LIB_PATH([cuda])
 if test "$with_cuda" != "no" ; then
+    echo "CPPFLAGS=$CPPFLAGS"
+    echo "CFLAGS=$CFLAGS"
+    echo "LDFLAGS=$LDFLAGS"
     PAC_CHECK_HEADER_LIB([cuda_runtime_api.h],[cudart],[cudaStreamSynchronize],[have_cuda=yes],[have_cuda=no])
     if test "${have_cuda}" = "yes" ; then
         AC_MSG_CHECKING([whether nvcc works])
         cat>conftest.cu<<EOF
         __global__ void foo(int x) {}
 EOF
-        ${with_cuda}/bin/nvcc -c conftest.cu 2> /dev/null
+        if test -n "$ac_save_CC" ; then
+            NVCC_FLAGS="-ccbin $ac_save_CC"
+            # if the host compiler is PGI, change the -ccbin to pgc++ instead of pgcc
+            NVCC_FLAGS=`echo $NVCC_FLAGS | sed 's/pgcc/pgc++/g'`
+        else
+            NVCC_FLAGS=""
+        fi
+        ${with_cuda}/bin/nvcc $NVCC_FLAGS -c conftest.cu 2> /dev/null
```

Comment:
Currently, it even finds CUDA via the environment except for the `nvcc` test if `with-cuda` does not contain a path. This would fix it for the test as well:

```suggestion
        # try nvcc from PATH if 'with-cuda' does not contain a valid path
        if test ${with_cuda} == "yes"; then
            nvcc_bin="nvcc"
        else
            nvcc_bin=${with_cuda}/bin/nvcc
        fi

        ${nvcc_bin} $NVCC_FLAGS -c conftest.cu 2> /dev/null
```